wandb/sdk/artifacts/artifact.py

```diff
@@ -1686,6 +1686,78 @@ def download(
         root = root or self._default_root()
         self._add_download_root(root)
 
+        if wandb.run is None:
+            with wandb.init(
+                entity=self._source_entity,
+                project=self._source_project,
+                job_type="auto",
+                settings=wandb.Settings(silent="true"),
+            ):
+                return self._run_artifact_download(
+                    root=root,
+                    recursive=recursive,
+                    allow_missing_references=allow_missing_references,
+                )
+        else:
+            return self._run_artifact_download(
+                root=root,
+                recursive=recursive,
+                allow_missing_references=allow_missing_references,
+            )
+
+    def _run_artifact_download(
+        self,
+        root: str,
+        recursive: bool = False,
+        allow_missing_references: bool = False,
+    ) -> FilePathStr:
+        assert wandb.run is not None, "failed to initialize run"
+        run = wandb.run
+        if run._settings._require_nexus:
+            if not run._backend or not run._backend.interface:
+                raise NotImplementedError
+            if run._settings._offline:
+                raise NotImplementedError("cannot download in offline mode")
+            # Start the download process in the user process too, to handle reference downloads
+            self._download(
```

Comment:
Move this below line 1732 to make the Python & Nexus side run in parallel instead of serially.

Reply:
Done already in https://github.com/wandb/wandb/pull/6535